### PR TITLE
Clear zero values from dataMap before Update validation

### PR DESCRIPTION
**Pull request by zimnx**
This is a workaround for issue which prohibits doing request validation in PreUpdate events using golang extensions. Structures provided to schema handlers which has fields with string type will be filled with empty string during allocation. When user doesn't provide this field in his request body, he don't intents to update it, but rather leave value as it is. Current code will try to update each of these fields to empty strings, which is not expected behavior. Additionally validation may fail because some fields may require that length is not 0.

### CODE REVIEW

**Comment by morrisson** [comment]
@zimnx It looks this PR may occur really bad side effect. If an user wants to update with "zero-value" such as `"", 0, false`, this code blocks to update those values. Can you confirm that?

**Comment by zimnx** [comment]
@morrisson You're right. None of our tests validated this use case. I've fixed implementation, and added missing testcase.

File: server/resources/resource_management_test.go [hunk 1]
```diff
@@ -1428,6 +1429,57 @@ var _ = Describe("Resource manager", func() {
...
+				Context("pre_update inserts zero-value", func() {
+					var (
+						requestMap map[string]interface{}
+					)
+					BeforeEach(func() {
+						requestMap = map[string]interface{}{
+							"test_string": "Steloj ne estas en ordo.",
+							"test_bool":   true,
+						}
+						events["pre_update"] = `if (context.resource.test_bool === undefined) { context.resource.test_bool = false; }`
```
Comment (michalzurawski) [inline server/resources/resource_management_test.go]: unnecessary

File: server/resources/resource_management.go [hunk 2]
```diff
@@ -24,6 +24,8 @@ import (
 	"strings"
 	"time"
 
+	"reflect"
```
Comment (michalzurawski) [inline server/resources/resource_management.go]: nit: goimport

File: server/resources/resource_management_test.go (hunk 1 above)
Comment (zimnx) [reply server/resources/resource_management_test.go]: why? It simulates Goext behavoir in JS code.

File: server/resources/resource_management.go [hunk 3]
```diff
@@ -795,6 +796,25 @@ func UpdateResourceInTransaction(
 		return ResourceError{err, "", Unauthorized}
 	}
 
+	if _, ok := context["go_validation"]; ok {
```
Comment (michalzurawski) [inline server/resources/resource_management.go]: "go_validation" appears already twice - maybe const?

Comment (zimnx) [reply server/resources/resource_management.go]: done

File: server/resources/resource_management.go (hunk 2 above)
Comment (zimnx) [reply server/resources/resource_management.go]: done

**Pull request merged**